### PR TITLE
unbundle argon2 from opendht

### PR DESCRIPTION
**Pull request by YogSottot**
fix https://github.com/YogSottot/ring-overlay/issues/6

### CODE REVIEW

File: net-libs/opendht/opendht-9999.ebuild [hunk 1]
```diff
@@ -42,7 +47,6 @@ src_configure() {
 }
 
 src_install() {
+	use !doc && rm README.md
 	cmake-utils_src_install
-
-	dodoc README.md
```
Comment (YogSottot) [inline net-libs/opendht/opendht-9999.ebuild]: no need for dodoc, it install it by default

File: net-libs/opendht/opendht-9999.ebuild [hunk 2]
```diff
@@ -24,14 +24,19 @@ LICENSE="GPL-3"
...
+DEPEND=">=app-crypt/argon2-20161029
+	>=dev-libs/msgpack-2.0
 	net-libs/gnutls
 	python? ( dev-python/cython[$(python_gen_usedep python3_{4,5,6})] )
 	tools? ( sys-libs/readline:0 )"
 RDEPEND="${DEPEND}"
 
+PATCHES=(
+	"${FILESDIR}/argon2_system.patch"
+)
```
Comment (YogSottot) [inline net-libs/opendht/opendht-9999.ebuild]: patch from debian https://packages.debian.org/source/sid/opendht

File: net-libs/opendht/opendht-9999.ebuild [hunk 3]
```diff
@@ -24,14 +24,19 @@ LICENSE="GPL-3"
 
 SLOT="0"
 
-IUSE="python static-libs tools"
+IUSE="doc python static-libs tools"
 
-DEPEND=">=dev-libs/msgpack-2.0
+DEPEND=">=app-crypt/argon2-20161029
+	>=dev-libs/msgpack-2.0
```
Comment (YogSottot) [inline net-libs/opendht/opendht-9999.ebuild]: in opendht now is bundled argon2-2016.10.25 https://github.com/savoirfairelinux/opendht/commit/28a5740c3754f6916416e7399c4b27b51a97928e

**Comment by stefan-langenmaier** [comment]
Thanks again. It's progresssing really good.

I added the patch also in the latest stable version of opendht. In the live ebuild I had to change the dependency on gnutls a little bit. I assume you built it with unstable version >=3.4.

**Comment by YogSottot** [comment]
Yes, i already switch to 3.5.9-r1

**Pull request merged**